src/screen/EditGifScreen.jsx

```diff
@@ -0,0 +1,340 @@
+      );
+
+      const data = await response.json();
+
+      if (data.result === "ng") {
+        console.log("실패");
+        return;
+      }
+
+      setGifUrl(data.file);
+      setIsLoading(false);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const saveGif = async () => {
+    try {
+      const filename = `${Date.now() + ".gif"}`;
+      const fileUri = `${FileSystem.documentDirectory}${filename}`;
+      const downloadedFile = await FileSystem.downloadAsync(gifUrl, fileUri);
+      const permission = await MediaLibrary.requestPermissionsAsync();
+
+      if (permission.status === "granted") {
+        setHasLibraryPermissions(permission.status === "granted");
+      }
+
+      if (permission.status !== "granted") {
+        navigation.navigate("Home");
+      }
+
+      const asset = await MediaLibrary.createAssetAsync(downloadedFile.uri);
+      const album = await MediaLibrary.getAlbumAsync("Download");
+
+      if (album === null) {
+        await MediaLibrary.createAlbumAsync("Download", asset, false);
+      } else {
+        await MediaLibrary.addAssetsToAlbumAsync([asset], album, false);
+      }
+
+      showToastMessage();
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const colorButtonPress = (title) => {
+    if (title === filter.color) {
+      setfilter({ ...filter, color: "" });
+    } else {
+      setfilter({ ...filter, color: title });
+    }
+  };
+
+  const gridButtonPress = (title) => {
+    if (title === filter.grid) {
+      setfilter({ ...filter, grid: "" });
+    } else {
+      setfilter({ ...filter, grid: title });
+    }
+  };
+
+  const fpsButtonPress = (title) => {
+    if (title === filter.fps) {
+      setfilter({ ...filter, fps: "" });
+    } else {
+      setfilter({ ...filter, fps: title });
+    }
+  };
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.header}>
+        <TouchableOpacity
+          onPress={() => navigation.goBack()}
+          style={styles.backButton}
+        >
+          <AntDesign name="left" size={27} color="black" />
+        </TouchableOpacity>
+      </View>
+      <View style={styles.contentsContainer}>
+        <Image
+          resizeMode="cover"
+          source={{
+            uri: gifUrl ? gifUrl : exampleGif,
+          }}
+          style={styles.mediaPreview}
+        />
+        <View style={styles.filtersContainer}>
+          <View style={styles.filterContainer}>
+            <Text style={styles.textTitle}>색상</Text>
+            <TouchableOpacity
+              style={{
+                ...styles.buttonFilter,
+                backgroundColor: filter.color === "SEPIA" ? "yellow" : "white",
+                borderWidth: filter.color === "SEPIA" ? "2px" : "1px",
+              }}
+              onPress={() => colorButtonPress("SEPIA")}
+            >
+              <Text style={styles.filter}>세피아</Text>
+            </TouchableOpacity>
+            <TouchableOpacity
+              style={{
```

Comment:
style 코드가 길어지면 가독성이 안좋아서 아래로 따로 빼는게 좋을 것 같아요
```suggestion
              style={sytles.button}
```